core/core.go

```diff
@@ -390,6 +404,41 @@ func (n *IpfsNode) loadBootstrapPeers() ([]peer.PeerInfo, error) {
 	return toPeerInfos(parsed), nil
 }
 
+func (n *IpfsNode) loadFilesRoot() error {
+	dsk := ds.NewKey("/filesroot")
+	pf := func(ctx context.Context, k key.Key) error {
+		return n.Repo.Datastore().Put(dsk, []byte(k))
+	}
+
+	var nd *merkledag.Node
+	val, err := n.Repo.Datastore().Get(dsk)
+
+	switch {
+	case err == ds.ErrNotFound || val == nil:
+		nd = &merkledag.Node{Data: unixfs.FolderPBData()}
```

Comment:
Should this be `uio.NewEmptyDirectory()`? as in https://github.com/ipfs/go-ipfs/blob/master/core/corehttp/gateway_handler.go#L277